EIPS/eip-1559.md

```diff
@@ -96,97 +93,87 @@ class Account:
 	code_hash: int = 0
 
 INITIAL_FORK_BLOCK_NUMBER = 10 # TBD
-MIGRATION_DURATION_IN_BLOCKS = 800000
-FINAL_FORK_BLOCK_NUMBER = INITIAL_FORK_BLOCK_NUMBER + MIGRATION_DURATION_IN_BLOCKS
 BASE_FEE_MAX_CHANGE_DENOMINATOR = 8
+ELASTICITY_MULTIPLIER = 2
 
 class World(ABC):
 	def validate_block(self, block: Block) -> None:
 		parent_base_fee = self.parent(block).base_fee
 		parent_gas_used = self.parent(block).gas_used
 		parent_gas_target = self.parent(block).gas_target
-		transactions_all = self.transactions(block)
+		transactions = self.transactions(block)
+
+		# check if the block used too much gas
+		assert block.gas_used <= block.gas_target * ELASTICITY_MULTIPLIER, 'invalid block: too much gas used'
 
 		# check if the block changed the gas target too much
 		assert block.gas_target <= parent_gas_target + parent_gas_target // 1024, 'invalid block: gas target increased too much'
 		assert block.gas_target >= parent_gas_target - parent_gas_target // 1024, 'invalid block: gas target decreased too much'
 
 		# check if the base fee is correct
-		if parent_gas_used >= parent_gas_target:
+		if parent_gas_used == parent_gas_target:
+			expected_base_fee = parent_base_fee
+		elif parent_gas_used > parent_gas_target:
 			gas_delta = parent_gas_used - parent_gas_target
 			fee_delta = max(parent_base_fee * gas_delta // parent_gas_target // BASE_FEE_MAX_CHANGE_DENOMINATOR, 1)
 			expected_base_fee = parent_base_fee + fee_delta
 		else:
 			gas_delta = parent_gas_target - parent_gas_used
-			fee_delta = (parent_base_fee * gas_delta - 1) // parent_gas_used // BASE_FEE_MAX_CHANGE_DENOMINATOR + 1
+			fee_delta = parent_base_fee * gas_delta // parent_gas_target // BASE_FEE_MAX_CHANGE_DENOMINATOR
 			expected_base_fee = parent_base_fee - fee_delta
```

Comment:
need to address cases when expected_base_fee ends up being < 0

Reply:
Discussed this in Discord, recording result here for thoroughness: In that particular conditional branch, there is no way for `gas_delta` to be negative, thus there is no way for `fee_delta` to be negative.  The same is true for the other branch.